src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1516,9 +1516,9 @@ class JoinOnConditionSegment(BaseSegment):
     type = "join_on_condition"
     match_grammar: Matchable = Sequence(
         "ON",
-        Indent,
+        Conditional(Indent, indented_on_contents=True),
         OptionallyBracketed(Ref("ExpressionSegment")),
-        Dedent,
+        Conditional(Dedent, indented_on_contents=True)
```

Comment:
Think you need the comma to make the linter happy:

```suggestion
        Conditional(Dedent, indented_on_contents=True),
```